include/zephyr/drivers/bluetooth/hci_driver.h

```diff
@@ -139,6 +139,14 @@ enum bt_hci_driver_bus {
 	BT_HCI_DRIVER_BUS_IPM           = 9,
 };
 
+#if defined(CONFIG_BT_HCI_SETUP) || defined(__DOXYGEN__)
+struct bt_hci_setup_params {
+#if defined(CONFIG_BT_HCI_SET_PUBLIC_ADDR)
+	bt_addr_t *public_addr;
```

Comment:
also why is it a pointer? It would probably be simpler to explicitly copy the address into the params.

Reply:
Sure, it's possible, but making it a pointer gives a convenient way of representing "no public address" as `NULL`.